File: python/order_details.py
#
# Example of retrieving a single order's details from the Marvelpress Order API
#
# This example makes use of the httplib2 python library to perform the GET request, similar results
# could also be achieved using python requests http://docs.python-requests.org/en/latest/index.html
# or other http libraries
# 
# Author: Oliver Smith 
# Copyright: Marvelpress Ltd. 2012
#

import urllib
import httplib2
import sys
import base64
import json

# Setup access credentials 
username  = "MP-00213/testing"
password  = "b21efa4cdacfd551e93e4b76ee7c3667"
url     = "https://orders.marvelpress.com/orders/order/"
orderId = "MP-00001-105"
thumbs = "1"

# Setup httplib2
http = httplib2.Http()
http.add_credentials(username, password)

# Setup and perform GET

# response consists of a tuple where [0] is the response object and [1]
# the document body
response = http.request(
    url+orderId+"/"+thumbs, 
    "GET", 
    headers = {'Content-type': 'application/x-www-form-urlencoded'}
)

#Check response status and show returned string
if response[0].status == 200:
    print "Continue OK!"
    print response[1]

else:
    print "Item continue failed"
    print response[1]

File: php/order_details.php
<?php
/**
 * Example of retrieving a single order's details from the Marvelpress Order API
 *
 * @author Lewis Pollard 
 * @copyright Marvelpress Ltd. 2012
 */

$url            = "https://orders.marvelpress.com/orders/order/";
$APIUsername    = "MP-00213/testing";                   
$APIPassword    = "b21efa4cdacfd551e93e4b76ee7c3667";
$order_id		= "MP-00001-86";
$thumbs			= TRUE;

//Initialise cURL
$ch = curl_init();

//Set the url and AUTH. cURL uses GET by default
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch,CURLOPT_URL,$url.$order_id."/".$thumbs);
curl_setopt($ch, CURLOPT_USERPWD, $APIUsername.':'.$APIPassword);

//Execute post and decode response
$result = json_decode(curl_exec($ch));

//Get http status
$http_status = curl_getinfo($ch , CURLINFO_HTTP_CODE);

curl_close($ch);

//Check if the request was successful 
if($http_status == 200)
{
    // Result now contains associative array of order details with a base64 encoded thumbnail image
    print_r($result);
}
else
{
    echo "Sending Order Failed \n";
    echo $result->Message."\n";
}